### ISSUE TITLE
Pyright fails to catch type error when passing attribute of generic type to method expecting the generic type

### ISSUE THREAD

**Issue by Naqu6** [microsoft/pyright#9232]
**Describe the bug**
Pyright (version 1.1.384) does not catch a type error when passing an attribute of a generic type to a method that expects the generic type itself. This occurs within a generic class where a method accepting type `T` is called with `t.attribute`, where `t` is of type `T` and `attribute` is a non-generic type (e.g., `int`).

**Code or Screenshots**

Here's a minimal repro:

```python
from dataclasses import dataclass
from typing import Generic, TypeVar

@dataclass
class Base:
    attribute: int

T = TypeVar('T', bound='Base')

class GenericClass(Generic[T]):
    def method_expecting_T(self, t: T):
        pass
    
    def method_with_error(self, t: T):
        self.method_expecting_T(t.attribute)  # This should be a type error, but Pyright doesn't catch it

    def correct_method(self, t: T):
        self.method_expecting_T(t)  # This is correct
```

```bash
$ pyright minimal.py 
0 errors, 0 warnings, 0 informations 
```

See https://github.com/microsoft/pyright/discussions/9231

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.385.
